pipeline/test_extract.py

```diff
@@ -0,0 +1,89 @@
+"""
+Tests the extraction file.
+"""
+import unittest
+from unittest.mock import patch, MagicMock
+
+from extract import get_domain_name, scrape_asos_page
+
+EXAMPLE_HTML_TEXT = '''
+<html><head><script>{"product_name":"Black Coat"}</script></head></html>'''
+EXAMPLE_SOUP_STRING = '''
+{"name":"Black Coat", "image": "http://asos.com/example_url", "productID": "123"}'''
+EXAMPLE_ASOS_URL = 'http://asos.com/blackcoats'
+ASOS_DOMAIN = 'asos.com'
+
```

Comment:
I like how you've set these as static variables - makes the script easier to read